src/wp-includes/class-wp-theme-json.php

```diff
@@ -2128,9 +2154,76 @@ protected static function compute_style_properties( $styles, $settings = array()
 			}
 		}
 
+		if ( $can_use_cached ) {
+			self::set_compute_style_properties_cache( $cache_key, $declarations );
+		}
+
 		return $declarations;
 	}
 
+	/**
+	 * Sets the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string $cache_key The cache key.
+	 * @param array  $declarations The declarations to cache.
+	 *
+	 * @return array The cached declarations.
+	 */
+	private static function set_compute_style_properties_cache( $cache_key, $declarations ) {
+		self::$cached_compute_style_properties[ $cache_key ] = $declarations;
+		set_site_transient( 'wp_compute_style_properties', self::$cached_compute_style_properties, HOUR_IN_SECONDS );
+		self::$cached_compute_style_properties;
+	}
+
+	/**
+	 * Gets the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return array The cached declarations.
+	 */
+	private static function get_compute_style_properties_cache() {
+		self::$cached_compute_style_properties = get_site_transient( 'wp_compute_style_properties' );
+
+		if ( ! is_array( self::$cached_compute_style_properties ) ) {
+			self::$cached_compute_style_properties = array();
+		}
+
+		return self::$cached_compute_style_properties;
+	}
+
+	/**
+	 * Clears the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 */
+	public static function clear_compute_style_properties_cache() {
```

Comment:
We would probably need to call this when other `WP_Theme_JSON` related caches are cleared, in `wp_clean_theme_json_cache()`.

Reply:
I did try that but it was breaking an old test case `test_get_user_data_from_wp_global_styles_does_not_use_uncached_queries_for_logged_out_users` where the count of queries executed was taken into consideration. But yes I can move call to clear_theme there.